src/Symfony/Component/Messenger/Transport/Doctrine/Connection.php

```diff
@@ -332,4 +344,9 @@ public static function formatDateTime(\DateTimeInterface $dateTime)
     {
         return $dateTime->format('Y-m-d\TH:i:s');
     }
+
+    private function decodeEnvelopeHeaders(array &$doctrineEnvelope): void
```

Comment:
What's the rationale for using references here? Can't it return another `array` instead? So you'd `return $this->decodeEnvelopeHeaders($data);` in `find` and map the `decodeEnvelopeHeaders` in `findAll`.

Reply:
Ah, I didn't think about mapping. I'll change it.